channel.go

```diff
@@ -276,6 +276,7 @@ func (ch *Channel) dispatch(msg message) {
 	case *channelClose:
 		ch.connection.closeChannel(ch, newError(m.ReplyCode, m.ReplyText))
 		ch.send(&channelCloseOk{})
+		ch.connection.releaseChannel(ch.id)
```

Comment:
Here we release channel ID after sending a `channel.close-ok`, which [causes channel process termination on the RabbitMQ end](https://github.com/rabbitmq/rabbitmq-server/blob/stable/src/rabbit_channel.erl#L879).

This is not consistent with what we do when the client initiates channel closure below.